gridstatus/ercot.py

```diff
@@ -432,42 +509,257 @@ def get_interconnection_queue(self, verbose=False):
 
         return queue
 
+    def get_spp(
+        self,
+        date,
+        end=None,
+        market: str = None,
+        locations: list = None,
+        location_type: str = None,
+        verbose=False,
+    ):
+        """Get SPP data for ERCOT
+
+        Supported Markets: REAL_TIME_15_MIN, DAY_AHEAD_HOURLY
+
+        Supported Location Types: "zone", "hub", "node"
+        """
+        if locations is None:
+            locations = "ALL"
+
+        if location_type is None:
+            location_type = LOCATION_TYPE_ZONE
+
+        assert market is not None, "market must be specified"
+        market = Markets(market)
+
+        if market == Markets.REAL_TIME_15_MIN:
+            return self._get_spp_rtm15(
+                date,
+                locations,
+                location_type,
+                verbose,
+            )
+        elif market == Markets.DAY_AHEAD_HOURLY:
+            return self._get_spp_dam(date, locations, location_type, verbose)
+        raise NotSupported(
+            f"Market {market} not supported for ERCOT",
+        )
+
+    def _get_spp_dam(
+        self,
+        date,
+        locations: list = None,
+        location_type: str = None,
+        verbose=False,
+    ):
+        """Get day-ahead hourly Market SPP data for ERCOT"""
+        today = pd.Timestamp.now(tz=self.default_timezone).normalize()
+        if date == "latest":
```

Comment:
just realized, there is probably no practical need to support `latest` for the DAM. That is because the entire day of prices is priced and released simultaneously. That means "latest" in the day ahead market will always mean the last hour of the day. This is different than the RTM which clears tick by tick. 

can we just remove support for latest? also remove from notebook example

Reply:
DAM is definitely interesting. What happens if we pass in `latest` for DAM then? Should it raise a ValueError or just return `today`?

Also if I wanted to get tomorrow's data today, I would pass in tomorrow's date explicitly?

Reply:
> DAM is definitely interesting. What happens if we pass in latest for DAM then? Should it raise a ValueError or just return today?

i think it should error for `latest`. the error message can say to use "today" instead. 

> DAM is definitely interesting. What happens if we pass in latest for DAM then? Should it raise a ValueError or just return today?
> 
> Also if I wanted to get tomorrow's data today, I would pass in tomorrow's date explicitly?

i need to do an audit across the ISOs of how dates are used to refer to day ahead prices and make sure grid status is consistent. I believe the most common approach is for the date to refer to when the energy is delivered. but because it is the day ahead market, the results are published the day before. 

I think this is what you are achieving with `publish_date = publish_date - pd.Timedelta("1D")`. 

So, the short answer is yes. You pass in tomorrows date.